test/test_flow_control_window.py

```diff
@@ -52,6 +52,32 @@ def test_flow_control_decreases_with_sent_data(self):
         remaining_length = self.DEFAULT_FLOW_WINDOW - len(b'some data')
         assert (c.local_flow_control_window(1) == remaining_length)
 
+    def test_flow_control_decreases_with_sent_data_with_padding(self):
+        """
+        When padded data is sent on a stream, the flow control window drops
+        by the length of the padding plus 1 for the 1-byte padding length
+        field.
+        """
+        c = h2.connection.H2Connection()
+        c.send_headers(1, self.example_request_headers)
+
+        c.send_data(1, b'some data', pad_length=5)
+        remaining_length = self.DEFAULT_FLOW_WINDOW - len(b'some data') - 6
+        assert c.local_flow_control_window(1) == remaining_length
+
+    def test_flow_control_decreases_with_sent_data_zero_length_padding(self):
+        """
+        When zero-length-padded data is sent on a stream, the flow control
+        window drops by the 1 for the 1-byte padding length field (which
+        is still present though zero-valued).
+        """
+        c = h2.connection.H2Connection()
+        c.send_headers(1, self.example_request_headers)
+
+        c.send_data(1, b'some data', pad_length=0)
+        remaining_length = self.DEFAULT_FLOW_WINDOW - len(b'some data') - 1
+        assert c.local_flow_control_window(1) == remaining_length
```

Comment:
These two tests are very similar. Could we combine them into two tests which take a `padding_length` parameter, and then define:

```python
remaining_length = self.DEFAULT_FLOW_WINDOW - len(b'some data') - padding_length - 1
```

?